vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -128,6 +130,41 @@ func (r *recommender) UpdateVPAs() {
 			klog.Errorf(
 				"Cannot update VPA %v/%v object. Reason: %+v", vpa.ID.Namespace, vpa.ID.VpaName, err)
 		}
+
+		lastUpdatedInfo := []map[string]string{}
+		containerStateMap := GetContainerNameToAggregateStateMap(vpa)
+		for container, aggregateState := range containerStateMap {
```

Comment:
Looks like we have somewhat copy-past of serialization and deserialization of freshness timestamp information for each: Checkpoint and VPA. We should factor out this code into a function and reuse the implementation.